operational_analysis/methods/plant_analysis.py

```diff
@@ -802,13 +847,50 @@ def filter_outliers(self, n):
             value_min=0.02 * plant_capac,
             value_max=1.2 * plant_capac,
         )
-
+        
+        if self.outlier_detection:
+            if self.time_resolution == 'M':
+                # Monthly linear regression (i.e., few data points): 
+                # filter outliers based on robust linear regression
+                # using Huber algorithm to flag outliers
+                X = sm.add_constant(df_sub[reanal])  # Reanalysis data with constant column
+                y = df_sub['gross_energy_gwh']*30/df_sub['num_days_expected']  # Energy data
+        
+                # Perform robust linear regression
+                rlm = sm.RLM(y, X, M=sm.robust.norms.HuberT(self._run.outlier_threshold))
+                rlm_results = rlm.fit()
+        
+                # Define valid data as points in which the Huber algorithm returned a value of 1
+                df_sub.loc[:, "flag_outliers"] = rlm_results.weights != 1
+            
+            else:
+                # Daily regressions (i.e., higher number of data points):
+                # Apply bin filter to catch outliers
+                df_sub.loc[:, "flag_outliers"] = filters.bin_filter(
+                        bin_col=df_sub["energy_gwh"],
```

Comment:
I remember that Mike had used gross_energy_gwh in the monthly (i.e., linear) case because gross energy has a stronger linear relationship with wind speed compared to energy_gwh. For the bin filter, I guess this is not relevant anymore, so I decided to apply the filter to energy_gwh to be consistent with the window filter, but I don't have a strong preference..

Reply:
Whatever you think is best. I mostly wanted to make sure this wasn't a typo. I just wonder if using the net energy instead of gross would remove too many points, since there could be curtailment or availability losses that contribute to outliers, but I don't think we'd necessarily want to remove them from the analysis, would we?

Reply:
Good point, I think you are right and it probably makes more sense to use gross energy here too. I'll change it.